### PR TITLE
Add option to use ThinLTO

### PR DESCRIPTION
**Pull request by Valentactive**
Nice to see support for the **LLD** linker was added to the master branch. Since I did id also last year in my own [fork](https://github.com/Valentactive/godot/commit/05564e527fad71f4558bac82f9abb50b4fd8b084) (although today rebased it on master) and added support for ThinLTO too, I thought maybe someone else may need it?
With this commit you only need to set 'use_thinlto=yes' and no need to set use_lto too (it will be set by this option itself for convenience) .

I only added it for Linux as I'm not sure how good ThinLTO works on other platforms (last year people said mixed things and I can't test on other platforms right now.).
Also ThinLTO only works when you use the LLD Linker so use_lld should be set too.

For more informations about ThinLTO see here:
[ThinLTO - Clang Documentation](https://clang.llvm.org/docs/ThinLTO.html)

### CODE REVIEW

File: SConstruct [hunk 1]
```diff
@@ -112,6 +112,7 @@ opts.Add(EnumVariable('target', "Compilation target", 'debug', ('debug', 'releas
 opts.Add(EnumVariable('optimize', "Optimization type", 'speed', ('speed', 'size')))
 opts.Add(BoolVariable('tools', "Build the tools (a.k.a. the Godot editor)", True))
 opts.Add(BoolVariable('use_lto', 'Use link-time optimization', False))
+opts.Add(BoolVariable('use_thinlto', 'Use ThinLTO, (no need to set use_lto too as this option will set it automatically)', False))
```
Comment (Calinou) [inline SConstruct]: I suggest shortening the description:

```suggestion
opts.Add(BoolVariable('use_thinlto', "Use ThinLTO (implies 'use_lto=yes')", False))
```

**Review by Calinou — CHANGES_REQUESTED** [review]

File: SConstruct (hunk 1 above)
Comment (Valentactive) [reply SConstruct]: Done :)

File: SConstruct [hunk 2]
```diff
@@ -112,6 +112,7 @@ opts.Add(EnumVariable('target', "Compilation target", 'debug', ('debug', 'releas
 opts.Add(EnumVariable('optimize', "Optimization type", 'speed', ('speed', 'size')))
 opts.Add(BoolVariable('tools', "Build the tools (a.k.a. the Godot editor)", True))
 opts.Add(BoolVariable('use_lto', 'Use link-time optimization', False))
+opts.Add(BoolVariable('use_thinlto', "Use ThinLTO (implies 'use_lto=yes')", False))
```
Comment (akien-mga) [inline SConstruct]: It shouldn't be added here if it's specific to one platform. Adding it in `platform/x11/detect.py` is sufficient.

**Review by akien-mga — CHANGES_REQUESTED** [review]

File: SConstruct (hunk 2 above)
Comment (Valentactive) [reply SConstruct]: Ok I removed it

**Review by akien-mga — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**